app/src/main/java/com/github/se/eventradar/viewmodel/EventsOverviewViewModel.kt

```diff
@@ -82,17 +112,41 @@ constructor(
   fun onViewListStatusChanged(state: MutableStateFlow<EventsOverviewUiState> = _uiState) {
     state.value = state.value.copy(viewList = !state.value.viewList)
   }
+
+  fun setFilterDialogOpen(isOpen: Boolean) {
+    _uiState.update { currentState -> currentState.copy(isFilterDialogOpen = isOpen) }
+  }
+
+  fun onSearchQueryChange(newQuery: String) {
+    _uiState.update { currentState -> currentState.copy(searchQuery = newQuery) }
+  }
+    /* TODO Christine
+    fun setRadiusInputFilter(radius: Double) {
+        _uiState.update { currentState ->
+            currentState.copy(radiusInputFilter = radius)
+        }
+    }
+    fun setFreeEventsFilter(isFree: Boolean) {
+        _uiState.update { currentState ->
+            currentState.copy(freeEventsFilter = isFree)
+        }
+    }
+
+     */
 }
 
 data class EventsOverviewUiState(
     val eventList: EventList = EventList(emptyList(), emptyList(), null),
-    var searchQuery: String = "",
-    var isFilterDialogOpen: Boolean = false,
+    val searchQuery: String = "",
+    val isFilterDialogOpen: Boolean = false,
+    // TODO: Christine, please update `radiusInputFilter` and `freeEventsFilter` to `val`.
```

Comment:
@christinehahaa Take a look at this please.

Reply:
Looks good, let's keep this in your PR since you've done tests for it already. That way the work is associated to you. 

I will change all my other UiState values I added to be val and not var.